### PR TITLE
EVA-3528: Refactor release tracker creation script 

### PR DESCRIPTION
**Pull request by apriltuesday**
* Remove clustering automation, this is entirely superseded by submission and assembly ingestion
* Move release tracker creation script to release automation
* Refactor to `ReleaseTracker` class

I've noted the only functional changes in the script, otherwise only refactoring.

### CODE REVIEW

File: eva-accession-release-automation/run_release_in_embassy/create_release_tracking_table.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+        query = f"""select distinct taxonomy_id as taxonomy, assembly_id as assembly_accession
+                    from evapro.supported_assembly_tracker"""
+        sources = 'DBSNP, EVA'
+        for tax, asm_acc in get_all_results_for_query(self.metadata_conn, query):
+            self._insert_entry_for_taxonomy_and_assembly(tax, asm_acc, sources)
+
+    def _insert_entry_for_taxonomy_and_assembly(self, tax, asm_acc, sources, sc_name=None, fasta_path=None,
+                                                report_path=None, release_folder_name=None):
+        sc_name = sc_name if sc_name else get_scientific_name_from_ensembl(tax)
+        sc_name = sc_name.replace("'", "\''")
```
Comment (apriltuesday) [inline eva-accession-release-automation/run_release_in_embassy/create_release_tracking_table.py:114]: change to accommodate our good friend `Ambystoma 'unisexual hybrid'`

File: eva-accession-release-automation/run_release_in_embassy/create_release_tracking_table.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+        return get_metadata_connection_handle(self.maven_profile, self.private_config_xml_file)
+
+    @cached_property
+    def mongo_conn(self):
+        mongo_uri = get_mongo_uri_for_eva_profile(self.maven_profile, self.private_config_xml_file)
+        return MongoDatabase(uri=mongo_uri, db_name="eva_accession_sharded")
+
+    def create_table_if_not_exists(self):
+        query_create_table = (
+            'create table if not exists eva_progress_tracker.clustering_release_tracker('
```
Comment (apriltuesday) [inline eva-accession-release-automation/run_release_in_embassy/create_release_tracking_table.py:54]: have left this table as-is but made a note of which fields we could get rid of

File: eva-accession-release-automation/run_release_in_embassy/create_release_tracking_table.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+        self.maven_profile = maven_profile
+        self.release_version = release_version
+        self.ref_dir = reference_directory
+
+    @cached_property
+    def metadata_conn(self):
+        return get_metadata_connection_handle(self.maven_profile, self.private_config_xml_file)
+
+    @cached_property
+    def mongo_conn(self):
```
Comment (apriltuesday) [inline eva-accession-release-automation/run_release_in_embassy/create_release_tracking_table.py:48]: Previously the mongo database being used was _always_ production, even if you were creating the table in `development` - this seemed messy/confusing to me so I've unified it to use a single profile, but it did have the advantage that you could create a table in dev based on real production data for `should_be_released`. I don't think it matters much for how we use the table now (e.g. we don't actually compute any counts that need to be checked), but let me know if you think this functionality should be restored.

File: eva-accession-release-automation/run_release_in_embassy/create_release_tracking_table.py (hunk 1 above)
Comment (apriltuesday) [reply eva-accession-release-automation/run_release_in_embassy/create_release_tracking_table.py:114]: Ticket to remove scientific name where it's not needed: [EVA-3405](https://www.ebi.ac.uk/panda/jira/browse/EVA-3405)

**Review by tcezard — APPROVED** [review]
Looks good and makes sense

**Pull request merged**